apps/src/randomPointGenerator.cpp

```diff
@@ -1,40 +1,76 @@
 #include <fstream>
 #include <iostream>
 #include <random>
-
 #include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
 
 #include "envitools/CSVIO.hpp"
 #include "envitools/EnviUtils.hpp"
 
+namespace fs = boost::filesystem;
+namespace po = boost::program_options;
+namespace et = envitools;
+
 // argv == et_PointGenerator xmin, ymin, xmax, ymax, total points, output path
 int main(int argc, char** argv) {
-    if (argc < 7) {
-        std::cout << " Usage: " << argv[0];
-        std::cout << " [xmin] [ymin] [xmax] [ymax] [total points] [output path]"
-                  << std::endl;
+    fs::path inputPath, csvPath;
+
+    // clang-format off
+    po::options_description options("Options");
+    options.add_options()
+        ("help,h", "Show this message")
+        ("input-file,i", po::value<std::string>()->required(),
+            "Input file path of ROIs")
+        ("output-file,o", po::value<std::string>()->required(),
+            "Output file path")
+        ("point-count,p", po::value<int>()->required(),
+            "total number of points to generate");
+    // clang-format on
+
+    // parsedOptions will hold the value of all parsed options as a Map
+    po::variables_map parsedOptions;
+    po::store(
+        po::command_line_parser(argc, argv).options(options).run(),
+        parsedOptions);
+
+    // show the help message
+    if (parsedOptions.count("help") || argc < 4) {
+        std::cout << options << std::endl;
+        return EXIT_SUCCESS;
+    }
+
+    // warn of missing options
+    try {
+        po::notify(parsedOptions);
+    } catch (po::error& e) {
+        std::cerr << "Error: " << e.what() << std::endl;
         return EXIT_FAILURE;
     }
+    
+    // Get the output path
+    csvPath = parsedOptions["output-file"].as<std::string>();
+    
+    // Get the ROI CSV file path
+    inputPath = parsedOptions["input-file"].as<std::string>();
+    
+    // Read ROIs into vector
+    auto vecROIs = et::CSVIO::ReadROICSV(inputPath);
 
-    int xmin = std::stoi(argv[1]);
-    int ymin = std::stoi(argv[2]);
-    int xmax = std::stoi(argv[3]);
-    int ymax = std::stoi(argv[4]);
-    int num_points = std::stoi(argv[5]);
+    // Get number of points to generate
+    int num_points = parsedOptions["point-count"].as<int>();
 
     std::cout << "X,Y" << std::endl;
     std::vector<cv::Vec2i> vec;
     for (int i = 0; i < num_points; i++) {
-        auto x = envitools::RandomInt(xmin, xmax);
-        auto y = envitools::RandomInt(ymin, ymax);
+        auto roi = vecROIs[envitools::RandomInt(0, vecROIs.size())];
```

Comment:
`size() - 1`. We should add documentation to the RandomInt() function in EnviUtils.hpp so that we don't forget that it's a closed interval [a, b].